AzureData/Source/AzureData.swift

```diff
@@ -148,12 +148,12 @@ public class AzureData {
     // MARK: - Collections
 
     // create
-    public static func create (collectionWithId collectionId: String, inDatabase databaseId: String, callback: @escaping (Response<DocumentCollection>) -> ()) {
-        return DocumentClient.shared.create (collectionWithId: collectionId, inDatabase: databaseId, callback: callback)
+    public static func create (collectionWithId collectionId: String, andPartitionKey partitionKey: DocumentCollection.PartitionKeyDefinition? = nil, inDatabase databaseId: String, callback: @escaping (Response<DocumentCollection>) -> ()) {
```

Comment:
```suggestion
    public static func create (collectionWithId collectionId: String, andPartitionKey partitionKey: DocumentCollection.PartitionKeyDefinition?, inDatabase databaseId: String, callback: @escaping (Response<DocumentCollection>) -> ()) {
```
I'm curious if we should put implicitly provide `nil` in these overrides.  I'm worried it'll cause confusion, and since we really want users to have to opt *out* of using partition keys, we should force them to provide `nil`.

I understand it'll likely be a breaking change, but we'll version appropriately and I think it'll cause less confusion/issues long term. 

I changed the ones I see, but it'll likely break some tests where one wasn't passed in.